### PR TITLE
[Bert] Feature: Custom Model Outputs

### LINKED ISSUE #22: Custom BERT Model outputs
For flexibility to fine-tune in downstream tasks, we should have the following options in the BERT family model outputs:

1. Return output of a pooling layer on top of [CLS] token embedding via a user configurable flag.(add_pooling_layer=True) like transformers API.
2. Return the full last hidden states of encoder layer : [Batch, Sequence, Embedding Dim] instead of just the [CLS] token embedding through a user configurable flag.

### PR DESCRIPTION
**Pull request by bkonkle**
Closes #22 

- Adds an optional Pooler layer for text classification using models like plain BERT instead of RoBERTa.
- Outputs both the last hidden states and the optional Pooler output if enabled.
- Adds the pooler layer to the loader, for using pretrained models.
- Makes the `pad_token_idx` public for things like the batcher to use.
- Removes `.clone()` in a few places where it isn't needed.
- Adds `.envrc` to gitignore for direnv users.
- Adds `.vscode` to gitignore for VS Code users.

### CODE REVIEW

**Review by nathanielsimard — APPROVED** [review]
LGTM ping @laggui for an additional review before merging.

File: bert-burn/src/embedding.rs [hunk 1]
```diff
@@ -18,13 +18,13 @@ pub struct BertEmbeddingsConfig {
 
 #[derive(Module, Debug)]
 pub struct BertEmbeddings<B: Backend> {
+    pub pad_token_idx: usize,
```
Comment (laggui) [inline bert-burn/src/embedding.rs:21]: Any particular reason why this is now public?

**Review by laggui — CHANGES_REQUESTED** [review]
Overall LGTM! One minor comment regarding implementation.

But I cloned your fork with the branch changes and tried to run the example with the same arguments as the README and it failed:

```
Model variant: roberta-base
thread 'main' panicked at src/loader.rs:304:56:
Config file present: InvalidFormat("missing field `with_pooling_layer` at line 21 column 1")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Seems the downloaded config does not have the newly added field and it fails to parse. We should fix that.

Have you tried the example with your changes @bkonkle?

**Comment by bkonkle** [comment]
Thanks for the review! I tested it, but then made the last-minute change to add `with_pooling_layer` to the config with, instead of passing it in as an argument. I failed to test it afterwards, and clearly forgot to wire the property in. :sweat_smile: I'll fix shortly.

File: bert-burn/src/embedding.rs (hunk 1 above)
Comment (bkonkle) [reply bert-burn/src/embedding.rs:21]: This is specifically so that I can use it here, in the batcher for my text classification pipeline (and later for the token classification pipeline): https://github.com/bkonkle/burn-transformers/blob/0.1.0/src/pipelines/text_classification/batcher.rs#L72

**Comment by bkonkle** [comment]
I went with `pub with_pooling_layer: Option<bool>` to avoid the problems with loading the base model config, coupled with `.unwrap_or(false)` to resolve the wrapped value.

Examples are working again for me:

```sh
Model variant: roberta-base
Input: Shape { dims: [3, 63] } // (Batch Size, Seq_len)
Roberta Sentence embedding Shape { dims: [3, 768] } // (Batch Size, Embedding_dim)

Model variant: bert-base-uncased
Input: Shape { dims: [3, 64] } // (Batch Size, Seq_len)
Roberta Sentence embedding Shape { dims: [3, 768] } // (Batch Size, Embedding_dim)

Model variant: roberta-large
KiB/s (0s)Input: Shape { dims: [3, 63] } // (Batch Size, Seq_len)
Roberta Sentence embedding Shape { dims: [3, 1024] } // (Batch Size, Embedding_dim)
```
